pages/resume.tsx

```diff
@@ -1,22 +1,83 @@
 import React, { FC } from 'react';
 import Resume from '../components/resume/resume';
-import jbatorski from '../data/resume/jbatorski.json';
+import { Basics, Job } from '../components/resume/types';
+import resumeJson from '../data/resume/resume.json';
+import { ResumeJson, WorkItem } from '../data/resume/types';
 
 interface ResumePageProps {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    data: any
+    basics: Basics;
+    skills: string[];
+    jobs: Job[];
 }
 
-const ResumePage: FC<ResumePageProps> = ({ data }) => {
-    return <Resume data={data} />;
+const ResumePage: FC<ResumePageProps> = uiModel => {
+    return <Resume {...uiModel} />;
 };
 
 export async function getStaticProps() {
+    const uiModel = transform(resumeJson as ResumeJson);
+
     return {
-        props: {
-            data: jbatorski,
+        props: uiModel,
+    };
+}
+
+export default ResumePage;
+
+function transform({ basics, skills, work }: ResumeJson) {
+    const uiModel = {
+        basics: {
+            name: basics.name,
+            email: basics.email,
+            url: basics.url,
+            summary: basics.summary,
         },
+        skills: skills.flatMap(skillSet => skillSet.keywords),
+        jobs: groupWorkItemsByCompany(work),
     };
+
+    return uiModel;
 }
 
-export default ResumePage;
\ No newline at end of file
+function groupWorkItemsByCompany(work: WorkItem[]) {
+    // Using map since it remembers insertion order
+    // so when converting back to array
+    // we don't need to worry bout sorting by job start/end times
+    const jobs = new Map<string, Job>();
+
+    work.forEach(workItem => {
+        let job: Job;
+
+        if (jobs.has(workItem.name)) {
+            job = jobs.get(workItem.name) as Job;
+        } else {
+            job = {
+                companyName: workItem.name,
+                startDate: workItem.startDate,
+                endDate: workItem.endDate || null,
```

Comment:
Had to change my types to be nullable and add these OR's to end dates since it looks like next.js throws errors if you try to return a property with an undefined value in `getServerSideProps`. See this [issue](https://github.com/vercel/next.js/discussions/11209) for a bit more context.